File: app/src/main/java/com/belov/artem/templateapp/presentation/view/fragment/BaseFragment.java
package com.belov.artem.templateapp.presentation.view.fragment;

import android.support.v4.app.Fragment;
import android.widget.Toast;


/**
 * Created by artem on 15.12.16.
 */

public class BaseFragment extends Fragment{

    @SuppressWarnings("unchecked")
    protected <C> C getComponent(Class<C> componentType) {
        return componentType.cast(((HasComponent<C>)getActivity()).getComponent());
    }

    protected void message(String message) {
        Toast.makeText(getActivity(), message, Toast.LENGTH_SHORT).show();
    }

}

File: app/src/main/java/com/belov/artem/templateapp/presentation/view/fragment/BaseFragment.kt
package com.belov.artem.templateapp.presentation.view.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import butterknife.ButterKnife
import com.trello.rxlifecycle.components.support.RxFragment

/**
 * Created by artem on 26.12.16.
 */
abstract class BaseFragment : RxFragment() {

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater?.inflate(layoutResource, container, false)
        if (view != null) {
            ButterKnife.bind(this, view)
        }
        return view
    }

    abstract val layoutResource: Int

    abstract fun onBackPressed()

    fun toast(resource: Int) {
        Toast.makeText(activity, resource, Toast.LENGTH_LONG).show()
    }

    fun toast(message: String) {
        Toast.makeText(activity, message, Toast.LENGTH_LONG).show()
    }
}